changes/1803.feature.md

```diff
@@ -0,0 +1 @@
+Add support for multi directory mkdir and fix type annotation accordingly
```

Comment:
This fragment will be used as a change log of release notes.
"What" we do here is kind of improving the feature of a CLI command, and that is what readers of release notes should know from this.
So, let's just inform readers the improvement and skip the "fixing type annotation" part which is "How". "How" should be in a description of PR.